Sources/DolbyIORTSUIKit/Private/Views/SingleStream/SingleStreamView.swift

```diff
@@ -14,29 +14,33 @@ struct SingleStreamView: View {
         static let offset: CGFloat = 200.0
     }
 
-    private let viewModel: SingleStreamViewModel
+    @ObservedObject var viewModel: StreamViewModel
```

Comment:
The idea is to not have child components dependant on parent ViewModel's and thus the component by itself can be reused and exposed public.

Reply:
`SingleStreamViewModel` doesn't listen for updates, so we only can use it as a ui state, not as a real viewModel (otherwise it will partly duplicate the logic in StreamViewModel).
`StreamViewModel` has been added here to get the updates.

Reply:
I thought `StreamingScreen` itself recompute its body on a state change in `StreamViewModel`, thus the single view also will recreated with a new SingleStreamViewModel ?

Also, we should focus on and let a user use SingleStreamViewModel by itself, so it should not be dependant on the parent view's view model. When you are free, can you show me the why doesn't the `SingleStreamViewModel` does not update on a state change in `StreamViewModel`?

Reply:
Did you get a chance to try this one @ldanilina ?

Reply:
Thanks a lot @aravind-raveendran , yes actually you're right, now I'm getting updates without getting `StreamViewModel.state`. Updated the PR.

Reply:
Thanks for checking !!